# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# DISTRHO Plugin Framework used by ZynAddSubFX really doesn't like Ardour's Lua
# environment, probably since it doesn't use plugin GUIs. This patch eliminates
# some strange errors that sometimes occur when loading or rendering Ardour
# projects in the Lua environment as a consequence of not initializing the GUI.

--- a/DPF/distrho/src/DistrhoPluginLV2.cpp
+++ b/DPF/distrho/src/DistrhoPluginLV2.cpp
@@ -685,7 +685,7 @@ public:
 
         updateParameterOutputsAndTriggers();
 
-#if DISTRHO_PLUGIN_WANT_STATE && DISTRHO_PLUGIN_HAS_UI
+#if false
         fEventsOutData.initIfNeeded(fURIDs.atomSequence);
 
         LV2_Atom_Event* aev;
